src/rateEngine/validators/_Validator.ts

```diff
@@ -21,7 +21,11 @@ abstract class Validator {
   }
 
   allErrors(): Array<Error> {
-    return this._errors.map(({errors}) => errors).flat();
+    // Ternary required because either TS or jest isn't able to flat map
+    // on the empty array for some reason ¯\_(ツ)_/¯
+    return this._errors.length > 0 ?
```

Comment:
i don't understand why this is necessary, but the test suite was giving me an error that `this._errors.map(...).flat()` was not a function when `this._errors` was the default, empty value. the default is just an empty array, which should be fine. maybe something within typescript causing havoc?

Reply:
That's really strange... I don't see any reason to worry about this fix, but needing it kinda makes me uneasy.

Reply:
same... i'm on Node 10.15.3 and tsc 4.0.5. maybe that version of Node is too old for fancy tricks like `.flat()`?